lib/nexmo_rack/verify_signature.rb

```diff
@@ -0,0 +1,40 @@
+# Verify Nexmo Signatures
+module Nexmo
+  module Rack
+    class VerifySignature
+      def initialize(app)
+        @app = app
+      end
+
+      def call(env)
+        req = ::Rack::Request.new(env)
+        if req.post?
+          params = req.params.dup 
+          verify = nexmo_client
+          if verify.check(params)
+            @app.call(env)
+          else
+            [403, {}, ['']]
+          end
+        else
+          @app.call(env)
+        end
+      end
+
+      private
+
+      def nexmo_client
+        if ENV['NEXMO_API_SIGNATURE']
+          verify = Nexmo::Signature.new(
+            ENV['NEXMO_API_SIGNATURE']
+          )
+        elsif defined?(Rails) && Rails.application.credentials.nexmo
+          verify = Nexmo::Signature.new(
+            Rails.application.credentials.nexmo[:api_signature]
+          )
```

Comment:
I wonder if we should add an else clause and raise, something like 'Missing configs' or something, we can drop the assignment as well.

Reply:
Adding the `else` is a good shout. Should we keep that in a separate PR and focus this one on the rename only with no behaviour change?

Reply:
definitely